state/machine.go

```diff
@@ -989,6 +962,120 @@ func (original *Machine) advanceLifecycle(life Life, force bool, maxWait time.Du
 }
 
+func (m *Machine) advanceLifecyleInitialOps(life Life) []txn.Op {
+	return []txn.Op{
+		{
+			C:      machinesC,
+			Id:     m.doc.DocID,
+			Update: bson.D{{"$set", bson.D{{"life", life}}}},
+		},
+		{
+			C:      machineUpgradeSeriesLocksC,
+			Id:     m.doc.Id,
+			Assert: txn.DocMissing,
+		},
+	}
+}
+
+func controllerAdvanceLifecyleVoteOp() txn.Op {
+	return txn.Op{
+		C:  controllersC,
+		Id: modelGlobalKey,
+		Assert: bson.D{
+			{"has-vote", bson.M{"$ne": true}},
+			{"wants-vote", bson.M{"$ne": true}},
+		},
+	}
+}
+
+// controllerIDsOp returns an Op to assert that the machine's
+// controllerIDs do not change.
+func (m *Machine) controllerIDsOp() (txn.Op, error) {
+	controllerIds, err := m.st.ControllerIds()
+	if err != nil {
+		return txn.Op{}, errors.Annotatef(err, "reading controller info")
+	}
+	if len(controllerIds) <= 1 {
+		return txn.Op{}, errors.Errorf("controller %s is the only controller", m.Id())
+	}
+	return txn.Op{
+		C:      controllersC,
+		Id:     modelGlobalKey,
+		Assert: bson.D{{"controller-ids", controllerIds}},
+	}, nil
+}
+
+// noContainersOp returns an Op to assert that the machine
+// has no containers.
+func (m *Machine) noContainersOp() txn.Op {
+	return txn.Op{
+		C:  containerRefsC,
+		Id: m.doc.DocID,
+		Assert: bson.D{{"$or", []bson.D{
+			{{"children", bson.D{{"$size", 0}}}},
+			{{"children", bson.D{{"$exists", false}}}},
+		}}},
+	}
+}
+
+// assessCanDieUnit returns true if the machine can die, based on
+// evaluating the provided unit.
+func (m *Machine) assessCanDieUnit(principalUnit string) (bool, error) {
+	canDie := true
+	u, err := m.st.Unit(principalUnit)
+	if err != nil {
+		return false, errors.Annotatef(err, "reading machine %s principal unit %v", m, m.doc.Principals[0])
+	}
+	app, err := u.Application()
+	if err != nil {
+		return false, errors.Annotatef(err, "reading machine %s principal unit application %v", m, u.doc.Application)
+	}
+	if u.Life() == Alive && app.Life() == Alive {
+		canDie = false
+	}
+	return canDie, nil
+}
+
+// noUnitAsserts returns bson DocElem which assert that there are
+// no units for the machine.
+func noUnitAsserts() bson.DocElem {
+	return bson.DocElem{
+		Name: "$or", Value: []bson.D{
+			{{"principals", bson.D{{"$size", 0}}}},
+			{{"principals", bson.D{{"$exists", false}}}},
+		},
+	}
+}
+
+// advanceLifecycleUnitAsserts returns bson DocElem which assert that there are
+// no units for the machine, or that the list of units has not changed.
+func advanceLifecycleUnitAsserts(principalUnitNames []string) bson.DocElem {
+	return bson.DocElem{
+		Name: "$or", Value: []bson.D{
+			{{Name: "principals", Value: principalUnitNames}},
+			{{Name: "principals", Value: bson.D{{"$size", 0}}}},
+			{{Name: "principals", Value: bson.D{{"$exists", false}}}},
+		},
+	}
+}
+
+// evaulateContainersAdvanceLifecycle determines if the machine has
+// containers, if so, returns the appropriate error.
+func (m *Machine) evaulateContainersAdvanceLifecycle() error {
```

Comment:
Misspelled evaluate. Maybe something like `advanceLifeCycleIfNoContainers` is more indicative of function...